workflow-ui/internal-testing-android/src/main/java/com/squareup/workflow1/ui/internal/test/Espresso.kt

```diff
@@ -0,0 +1,44 @@
+package com.squareup.workflow1.ui.internal.test
+
+import android.view.View
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.Root
+import androidx.test.espresso.ViewInteraction
+import androidx.test.espresso.action.ViewActions
+import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.isRoot
+import org.hamcrest.Description
+import org.hamcrest.Matcher
+import org.hamcrest.Matchers.allOf
+import org.hamcrest.Matchers.any
+import org.hamcrest.TypeSafeMatcher
+
+/**
+ * Fork of [Espresso.onView] that looks in all [Root]s, not just the one matched by the default
+ * root matcher. The default root matcher will sometimes fail to find views in our dialogs.
+ */
+public fun onWorkflowView(viewMatcher: Matcher<View>): ViewInteraction {
```

Comment:
Drive by bikeshed: there's really nothing workflow specific about either of these methods. I'd suggest `inAnyRoot()` and `pressBackThatActuallyFuckingWorks()`.

Reply:
I like `inAnyRoot` much better, how about `actuallyPressBack` for the other?

Reply:
#308